syne_tune/optimizer/schedulers/searchers/bore/multi_fidelity_bore.py

```diff
@@ -15,13 +15,16 @@
 import numpy as np
 
 from syne_tune.optimizer.schedulers.searchers.bore import Bore
+from syne_tune.optimizer.schedulers.searchers.bracket_searcher import (
+    DefaultHyperbandBracketSamplingSearcher,
+)
 
 __all__ = ["MultiFidelityBore"]
 
 logger = logging.getLogger(__name__)
 
 
-class MultiFidelityBore(Bore):
+class MultiFidelityBore(Bore, DefaultHyperbandBracketSamplingSearcher):
```

Comment:
do we use multi-inheritance now? please be careful with this, calling super.init is not trivial based on my experience.

Reply:
Do you know a safer alternative? I'd like special searchers to provide another method `distribution_over_brackets`, this is done in `SearcherWithDistributionOverBrackets`. Then, I provide some default implementation in `DefaultHyperbandBracketSamplingSearcher`, which ends being used in quite some searchers.

If this can be done better, please let me know.

But I've been quite careful, in that all classes involved along the `SearcherWithDistributionOverBrackets` always just have the same `__init__` signature as the base class, and just call the super class init. This seems to me to avoid pitfalls.

Reply:
based on my understanding of your code, you might be able to use a delegator pattern instead

Reply:
OK I see, this would be another class just for the purpose of computing this distribution. I can do it that way, I think.

Reply:
Not sure you need another class. You might just create an instance of the one you have right now and call its function instead.